### PR TITLE
Reset account

### PR DESCRIPTION
**Pull request by j0k3r**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | yes
| Translation   | yes
| Fixed tickets | https://github.com/wallabag/wallabag/issues/1507
| License       | MIT

Should fix https://github.com/wallabag/wallabag/issues/1507

Ability to remove:
- all annotations
- all tags
- all entries (which means also all tags & annotations)

Preview in material theme:

Preview in baggy theme:

Also, the emoji insertion in MySQL should be fixed to (fixing https://github.com/wallabag/wallabag/issues/2468).
We switched on charset `utf8mb4` to fix that.

- [x] add migration script for utf8mb4 table (https://mathiasbynens.be/notes/mysql-utf8mb4#utf8-to-utf8mb4)

### CODE REVIEW

File: src/Wallabag/AnnotationBundle/Repository/AnnotationRepository.php [hunk 1]
```diff
@@ -106,4 +106,17 @@ public function findOneByUsername($username)
...
+    /**
+     * Remove all annotations for a user id.
+     * Used when a user want to reset all informations.
+     *
+     * @param int $userId
+     */
+    public function removeAllByUserId($userId)
+    {
+        $this->getEntityManager()
+            ->createQuery('DELETE FROM Wallabag\AnnotationBundle\Entity\Annotation a WHERE a.user = '.$userId)
```
Comment (tcitworld) [inline src/Wallabag/AnnotationBundle/Repository/AnnotationRepository.php]: Maybe a prepared statement instead ?

File: src/Wallabag/UserBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ services:
         arguments:
             - WallabagUserBundle:User
 
-    wallabag_user.create_config:
+    wallabag_user.listener.create_config:
```
Comment (tcitworld) [inline src/Wallabag/UserBundle/Resources/config/services.yml]: ooooh !

**Comment by tcitworld** [comment]
> all entries (which means also all tags & annotations)

This should be told to the user

File: app/config/parameters.yml.dist [hunk 3]
```diff
@@ -19,16 +19,18 @@ parameters:
     database_path: "%kernel.root_dir%/../data/db/wallabag.sqlite"
     database_table_prefix: wallabag_
     database_socket: null
+    # with MySQL, use "utf8mb4" if got problem with content with emojis
+    database_charset: utf8
```
Comment (nicosomb) [inline app/config/parameters.yml.dist]: You set `utf8` as a default value (and the default database storage is SQLite) but you put `utf8mb4` in `parameters_test.sqlite.yml`. Are you sure?.

File: src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml [hunk 4]
```diff
@@ -88,6 +88,13 @@ config:
         name_label: 'Nom'
         email_label: 'Adresse e-mail'
         twoFactorAuthentication_label: 'Double authentification'
+    reset:
+        title: Réinitialisation (attention danger !)
+        description: En cliquant sur les boutons ci-dessous vous avez la possibilité de supprimer certaines informations de votre compte. Attention, ces actions sont IRREVERSIBLES !
```
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml]: IRRÉVERSIBLES

File: src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml [hunk 5]
```diff
@@ -88,6 +88,13 @@ config:
...
+        annotations: Supprimer TOUTES les annotations
+        tags: Supprimer TOUS les tags
+        entries: Supprimer TOUS les articles
+        confirm: Êtes-vous vraiment vraiment sûr? (C'EST IRREVERSIBLE)
```
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml]: `sûr ?` and IRRÉVERSIBLE

File: src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml [hunk 6]
```diff
@@ -457,6 +464,9 @@ flashes:
             tagging_rules_deleted: 'Règle supprimée'
             user_added: 'Utilisateur "%username%" ajouté'
             rss_token_updated: 'Jeton RSS mis à jour'
+            annotations_reset: Annotations supprimées
+            tags_reset: Tags supprimés
+            entries_reset: Articles supprimés
     entry:
         notice:
             entry_already_saved: 'Article déjà sauvergardé le %date%'
```
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/translations/messages.fr.yml]: ho, a typo! `sauvergardé` -> `sauvegardé`

**Comment by nicosomb** [comment]
It works, even with SQLite 👍

**Pull request merged**